server/serverAPI/controller/InventoryController.ts

```diff
@@ -109,41 +157,54 @@ export default class InventoryController extends BaseUserController {
     update = async (req: Request, res: Response) => {
         let parameters = new Map<string, any>([["username", req.serverUser.username]]);
 
-        return this.requestGet(parameters, res).then(user => {
-            let isFound: boolean = false;
+        try {
+            let user = await this.requestGet(parameters, res);
 
-            let newInventory: IInventoryIngredient[] = [];
+            let isFound: boolean = false;
 
             for (let i = 0; i < user.inventory.length; i++) {
-                let ingredientToAdd = user.inventory[i];
+                let existingIngredient = user.inventory[i];
 
                 if (user.inventory[i].id === Number.parseInt(req.params.foodID)) {
                     isFound = true;
 
-                    ingredientToAdd = {
-                        id: Number.parseInt(req.params.foodID),
-                        name: this.isStringUndefinedOrEmpty(req.body.name) ? ingredientToAdd.name : req.body.name,
-                        category: this.isStringUndefinedOrEmpty(req.body.category) ? ingredientToAdd.category : req.body.category,
-                        nutrients: this.isStringUndefinedOrEmpty(req.body.nutrients) ?
-                            ingredientToAdd.nutrients : this.parseNutrients(req.body.nutrients),
-                        expirationDate: this.isStringUndefinedOrEmpty(req.body.expirationDate) ?
-                            ingredientToAdd.expirationDate : Number.parseFloat(req.body.expirationDate)
-                    };
+                    user.inventory[i] = await this.parseUpdateRequest(req, res, existingIngredient);
+
+                    if (user.inventory[i].quantity?.unit !== existingIngredient.quantity?.unit ||
+                        user.inventory[i].quantity?.value !== existingIngredient.quantity?.value) {
+
+                        let updatedIngredient = await this.foodAPI.Get(
+                            new Map<string, any>([
+                                ["id", existingIngredient.id],
+                                ["quantity", user.inventory[i].quantity!.value],
+                                ["unit", user.inventory[i].quantity!.unit]
+                            ])
+                        );
```

Comment:
In your test cases, have you handled the cases:
1) `user.inventory[i].quantity?.unit` is null/undefined and existingIngredient.quantity? is not null/undefined?
2) `user.inventory[i].quantity?.unit` is not null/undefined and existingIngredient.quantity? is null/undefined?
I see you use `user.inventory[i].quantity!` and I was wondering if that would have a problem in either of the two cases above.

Reply:
parseUpdateRequest takes that into consideration, it checks quantitySchema and throws if there is a problem. Tests are yet to be created but it works :+1: